src/main/java/com/merkrafter/lexing/CharTokenizer.kt

```diff
@@ -22,6 +25,11 @@ class CharTokenizer(private val input: Sequence<Char>) : Iterator<Token> {
      */
     private val inputIterator = input.iterator()
 
+    /**
+     * The last read character
+     */
+    private var ch: Char = ' '
```

Comment:
I'm not satisfied with this field yet.
First, using a space character to signal "uninitialized" is not suitable, especially because the Tokenizer should be able to handle whitespace as normal characters later on.
Second, a more general approach is needed to store characters and retrieve them. This is necessary for token boarders (if, for instance, after some digits a `+` occurs this `+` must be stored somewhere so the next tokenizing method can use it). Hence, all `inputTokenizer.next()` calls should probably be replaced with a dedicated `nextCharacter` method that checks whether some character is currently stored and retrieves it or otherwise reads the next one from the underlying character iterator.

Reply:
This was now implemented using a Queue<Char>